Added support for lhco.gz files

adarshp opened the pull request:
When you run the Madgraph-Pythia-Delphes event generation chain, Delphes runs root2lhco on the final root files, and then compresses them. The final LHCO files are zipped and will have names of the form tag_1_delphes_events.lhco.gz. This change enables LHCO_reader to process them.

adarshp committed:
Added support for lhco.gz files

When you run the Madgraph-Pythia-Delphes event generation chain, Delphes runs root2lhco on the final root files, and then compresses them. The final LHCO files are zipped and will have names of the form tag_1_delphes_events.lhco.gz. This change enables LHCO_reader to process them.

adarshp committed:
Merge pull request #1 from adarshp/gzip-file-support

Added support for lhco.gz files

Commit:
fixed indentation issues

innisfree commented:
Thanks a lot! This is a great idea. I didn't know it was so easy with the gzip package.

I've implemented this in 36de0b83962030823fa40e97fc11c01fe4c72527.

I didn't want to pull your request as it is because there's a a big repetition of a block of code for `with gunzip.open(` and `with open(...`. I've gone with

```
if self.zipfile:
    open_ = gzip.open
else:
    open_ = open

with open_(...
```
but the idea is identical. I've also added a bit more careful checking about what file-names are supplied - *.root.gz isn't accepted, only *.lhco.gz. I'll include this all in the next release after a bit more testing.

Thanks again.

adarshp commented:
No problem. Also, thanks for writing this package and documenting it so well - it is making writing analyses a lot easier!

-Adarsh

> On Mar 31, 2016, at 4:42 PM, Andrew Fowlie <notifications@github.com> wrote:
> 
> Thanks a lot! This is a great idea. I didn't know it was so easy with the gzip package.
> 
> I've implemented this in 36de0b8 <https://github.com/innisfree/LHCO_reader/commit/36de0b83962030823fa40e97fc11c01fe4c72527>.
> 
> I didn't want to pull your request as it is because there's a a big repetition of a block of code for with gunzip.open( and with open(.... I've gone with
> 
> if self.zipfile:
>     open_ = gzip.open
> else:
>     open_ = open
> 
> with open_(...
> but the idea is identical. I've also added a bit more careful checking about what file-names are supplied - *.root.gz isn't accepted, only *.lhco.gz. I'll include this all in the next release after a bit more testing.
> 
> Thanks again.
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub <https://github.com/innisfree/LHCO_reader/pull/9#issuecomment-204176662>
